File: app/src/main/java/com/example/android/popularmovies/database/VideoUrlsConverter.java
package com.example.android.popularmovies.database;

import androidx.room.TypeConverter;

class VideoUrlsConverter {

    @TypeConverter
    public static String toString(String[] arr){
        StringBuilder builder = new StringBuilder();
        for(String str : arr){
            builder.append(str).append("\n");
        }
        return  builder.toString();
    }

    @TypeConverter
    public static String[] toStringArray(String str){
        if(str.isEmpty()){
            return new String[0];
        }
        return str.split("\n");
    }


}

File: app/src/main/java/com/example/android/popularmovies/database/VideoUrlsConverter.kt
package com.example.android.popularmovies.database

import androidx.room.TypeConverter

class VideoUrlsConverter {

    @TypeConverter
    fun toString(arr: Array<String?>): String {
        val builder = StringBuilder()
        for (str in arr) {
            builder.append(str).append("\n")
        }
        return builder.toString()
    }

    @TypeConverter
    fun toStringArray(str: String): Array<String?> {
        return if (str.isEmpty()) {
            arrayOfNulls(0)
        } else str.split("\n").toTypedArray()
    }
}